### PR TITLE
Adding Support for BaasBucketName to Snapshot Restore Functionality

### PR DESCRIPTION
**Pull request by cdchris12**
This PR adds functionality to the snapshot restore functionality such that it is aware of the newly introduced `LAGOON_BAAS_BUCKET_NAME` in order to allow snapshot restores to complete successfully for migrated projects.

# Checklist

- [ ] Affected Issues have been mentioned in the Closing issues section
- [ ] Documentation has been written/updated
- [X] PR title is ready for changelog and subsystem label(s) applied

# Closing issues

This PR resolves AMAZEEIO-3633

### CODE REVIEW

File: node-packages/commons/src/tasks.ts [hunk 1]
```diff
@@ -1038,7 +1038,7 @@ const restoreConfig = (name, backupId, safeProjectName) => {
       },
       backend: {
         s3: {
-          bucket: `baas-${safeProjectName}`
+          bucket: `${baasBucketName ? baasBucketName : "baas-".concat(safeProjectName)}`
```
Comment (Schnitzel) [inline node-packages/commons/src/tasks.ts]:
```suggestion
          bucket: baasBucketName ? baasBucketName : `baas-${safeProjectName}`
```

File: node-packages/commons/src/tasks.ts [hunk 2]
```diff
@@ -1148,7 +1148,15 @@ export const createMiscTask = async function(taskData: any) {
...
-          const restoreConf = restoreConfig(restoreName, taskData.data.backup.backupId, makeSafe(taskData.data.project.name))
+
+          let baasBucketName = ""
+          for (let variable of osResult.envVariables) {
+            if (variable.name == "LAGOON_BAAS_BUCKET_NAME") {
+              baasBucketName = variable.value
+            }
+          }
+          // Parse out the baasBucketName for any migrated projects
+          const restoreConf = restoreConfig(restoreName, taskData.data.backup.backupId, makeSafe(taskData.data.project.name), makeSafe(baasBucketName))
```
Comment (Schnitzel) [inline node-packages/commons/src/tasks.ts]: do we need to run `makeSafe()` on the `baasBucketName`? this will change the `baasBucketName` with replacing an `_` to a `-` which could cause the restore to fail as the backup system maybe actually used a `_`

File: services/kubernetesmisc/src/handlers/resticRestore.ts [hunk 3]
```diff
@@ -60,7 +60,7 @@ const restoreConfig = (name, backupId, safeProjectName) => {
       },
       backend: {
         s3: {
-          bucket: `baas-${safeProjectName}`
+          bucket: `${baasBucketName ? baasBucketName : "baas-".concat(safeProjectName)}`
```
Comment (Schnitzel) [inline services/kubernetesmisc/src/handlers/resticRestore.ts]:
```suggestion
          bucket: baasBucketName ? baasBucketName : `baas-${safeProjectName}`
```

File: services/openshiftmisc/src/handlers/resticRestore.ts [hunk 4]
```diff
@@ -51,7 +51,7 @@ export async function resticRestore (data) {
         },
         backend: {
           s3: {
-            bucket: `baas-${safeProjectName}`
+            bucket: `${baasBucketName ? baasBucketName : "baas-".concat(safeProjectName)}`
```
Comment (Schnitzel) [inline services/openshiftmisc/src/handlers/resticRestore.ts]:
```suggestion
            bucket: baasBucketName ? baasBucketName : `baas-${safeProjectName}`
```

**Review by Schnitzel — CHANGES_REQUESTED** [review]

File: node-packages/commons/src/tasks.ts (hunk 2 above)
Comment (cdchris12) [reply node-packages/commons/src/tasks.ts]: We don't _need_ to run that here, no. I'll update accordingly.

File: node-packages/commons/src/tasks.ts [hunk 5]
```diff
@@ -1148,7 +1148,15 @@ export const createMiscTask = async function(taskData: any) {
           // Handle setting up the configuration for a restic restoration task
           const restoreName = `restore-${R.slice(0, 7, taskData.data.backup.backupId)}`;
           // generate the restore CRD
-          const restoreConf = restoreConfig(restoreName, taskData.data.backup.backupId, makeSafe(taskData.data.project.name))
+
+          let baasBucketName = ""
+          for (let variable of osResult.envVariables) {
```
Comment (rocketeerbkw) [inline node-packages/commons/src/tasks.ts]: The result of `getOpenShiftInfoForProject` is an object keyed by `project` so you'd want to do `osResult.project.envVariables` or use `osProjectOpenShift.envVariables`.

File: node-packages/commons/src/tasks.ts [hunk 6]
```diff
@@ -1228,7 +1236,16 @@ export const createMiscTask = async function(taskData: any) {
           // Handle setting up the configuration for a restic restoration task
           const restoreName = `restore-${R.slice(0, 7, taskData.data.backup.backupId)}`;
           // generate the restore CRD
-          const restoreConf = restoreConfig(restoreName, taskData.data.backup.backupId, makeSafe(taskData.data.project.name))
+
+          let baasBucketName = ""
+          for (let variable of result.envVariables) {
```
Comment (rocketeerbkw) [inline node-packages/commons/src/tasks.ts]: The result of `getOpenShiftInfoForProject` is an object keyed by `project` so you'd want to do `result.project.envVariables` or use `projectOpenShift.envVariables`.

File: services/kubernetesmisc/src/handlers/resticRestore.ts [hunk 7]
```diff
@@ -86,8 +86,16 @@ async function resticRestore (data: any) {
 
   try {
 
+    let baasBucketName = ""
+    for (let variable of project.projectInfo.envVariables) {
```
Comment (rocketeerbkw) [inline services/kubernetesmisc/src/handlers/resticRestore.ts]: The result of `getOpenShiftInfoForProject` is an object keyed by `project`. The [destructuring](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment#Object_destructuring) on line 78 creates a local variable `projectInfo` with the contents of `project` so you'd want to use `projectInfo.envVariables` here.

File: services/openshiftmisc/src/handlers/resticRestore.ts [hunk 8]
```diff
@@ -75,11 +75,19 @@ export async function resticRestore (data) {
   });
 
   try {
+    let baasBucketName = ""
+    for (let variable of result.envVariables) {
```
Comment (rocketeerbkw) [inline services/openshiftmisc/src/handlers/resticRestore.ts]: The result of `getOpenShiftInfoForProject` is an object keyed by `project` so you'd want to do `result.project.envVariables` or use `projectOpenShift.envVariables`.

File: node-packages/commons/src/tasks.ts [hunk 9]
```diff
@@ -1148,7 +1148,15 @@ export const createMiscTask = async function(taskData: any) {
...
           // generate the restore CRD
-          const restoreConf = restoreConfig(restoreName, taskData.data.backup.backupId, makeSafe(taskData.data.project.name))
+
+          let baasBucketName = ""
+          for (let variable of osResult.envVariables) {
+            if (variable.name == "LAGOON_BAAS_BUCKET_NAME") {
+              baasBucketName = variable.value
+            }
+          }
+          // Parse out the baasBucketName for any migrated projects
```
Comment (rocketeerbkw) [inline node-packages/commons/src/tasks.ts]: Is this a comment related to this PR? migrations weren't mentioned in the description. If so, I'd put this above this block of code. Same comment for all the other instances of this comment.

**Review by rocketeerbkw — CHANGES_REQUESTED** [review]
Is there a specific reason to use a loop for setting `baasBucketName`? Since the name of environment variables is unique per project, you can also do a `find` or `reduce`. At a minimum, you can `break` the loop as soon as `LAGOON_BAAS_BUCKET_NAME` is found.

**Review by rocketeerbkw — APPROVED** [review]
